trlx/trainer/accelerate_ppo_trainer.py

```diff
@@ -470,18 +470,21 @@ def make_experience(self, num_rollouts: int = 1024, iter_count: int = 0):  # noq
                 )
 
                 rollout_count += 1
-            exp_time = clock.tick()
+
+            if torch.distributed.is_initialized():
+                torch.distributed.all_reduce(mean_kl, torch.distributed.ReduceOp.AVG)
+
+            stats["time/rollout_time"] = clock.tick()
+            stats["policy/sqrt_kl"] = torch.sqrt(mean_kl).item()
+            accumulated_stats.append(stats)
+
             tbar.set_description(f"[rollout {len(ppo_rl_elements)} / {num_rollouts}]")
             tbar.update(min(rollout_count, num_rollouts))
         tbar.close()
 
-        if torch.distributed.is_initialized():
-            torch.distributed.all_reduce(self.mean_kl, torch.distributed.ReduceOp.AVG)
-
-        stats["policy/sqrt_kl"] = torch.sqrt(self.mean_kl).item()
+        stats = {k: sum([xs[k] for xs in accumulated_stats]) / len(accumulated_stats) for k in stats}
```

Comment:
Strictly speaking this isn't necessarily correct e.g. if we are recording the max over all local rollouts. However I don't know how to perform the correct reduction without annotating each stat, so this seems fine for now

Reply:
There are no maxs/mins in this particular dict, but I see what you are saying. Also I guess you're referring to https://github.com/CarperAI/trlx/blob/9bc08369ca9ec83342c4d7755205dab1a7723006/trlx/trainer/accelerate_base_trainer.py#L544 but it's also pretty easy to fix, perhaps in a separate PR